src/api.rs

```diff
@@ -7,6 +7,93 @@ use crate::futures::market::*;
 use crate::userstream::*;
 use crate::client::*;
 
+pub enum API {
+    Spot(Spot),
+    Futures(Futures),
+}
+
+/// Endpoint for production and test orders.
+///
+/// Orders issued to test are validated, but not sent into the matching engine.
+pub enum Spot {
+    Ping,
+    Time,
+    ExchangeInfo,
+    Order,
+    OrderTest,
+    Account,
+    OpenOrders,
+    MyTrades,
+    Depth,
+    Price,
+    AvgPrice,
+    BookTicker,
+    Ticker24hr,
+    Klines,
+    UserDataStream,
+}
+
+pub enum Futures {
+    Ping,
+    Time,
+    ExchangeInfo,
+    Trades,
+    HistoricalTrades,
+    AggTrades,
+    Depth,
+    BookTicker,
+    Ticker24hr,
+    TickerPrice,
+    Klines,
+    PremiumIndex,
+    AllForceOrders,
+    OpenInterest,
+}
+
+impl From<API> for String {
+    fn from(item: API) -> Self {
+        match item {
+            API::Spot(route) => {
+                match route {
+                    Spot::Ping => String::from("/api/v3/ping"),
```

Comment:
I didn't try but you could pull out the `String::from(` at each line and put it at line 54, e.g. change line 54 into
`String::from( match item {`